ui/src/lib.rs

```diff
@@ -73,31 +76,38 @@ fn update(msg: Msg, model: &mut Model, orders: &mut impl Orders<Msg>) {
         Msg::RouteChanged(maybe_route) => {
             handle_route(maybe_route, model);
         }
-        Msg::StartGameMsg(sub_msg) => {
+        Msg::Rendered(render_info) => {
+            model
+                .get_session_mut()
+                .set_current_time(render_info.timestamp);
+        }
+        Msg::StartGame(sub_msg) => {
             if let Model::StartGame(sub_model) = model {
-                page::start_game::update(sub_msg, sub_model, &mut orders.proxy(Msg::StartGameMsg))
+                page::start_game::update(sub_msg, sub_model, &mut orders.proxy(Msg::StartGame))
             }
         }
-        Msg::DemoMsg(sub_msg) => {
+        Msg::Demo(sub_msg) => {
             if let Model::Demo(_, sub_model) = model {
                 page::demo::update(sub_msg, sub_model)
             }
         }
-        Msg::LobbyMsg(sub_msg) => {
+        Msg::Lobby(sub_msg) => {
             if let Model::Lobby(sub_model) = model {
                 page::lobby::update(sub_msg, sub_model)
             }
         }
-        Msg::GameMsg(sub_msg) => {
+        Msg::Game(sub_msg) => {
             if let Model::Game(sub_model) = model {
                 page::game::update(sub_msg, sub_model)
             }
         }
     }
+
+    orders.after_next_render(Msg::Rendered);
```

Comment:
This is an important part. It reads like this "After every time we render the application, pass in this `Msg` with information about when the last render happened"